ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/transaction/arjunacore/TransactionImple.java

```diff
@@ -1032,6 +1032,40 @@ public String toString()
 		}
 	}
 
+	public void endAssociation(Xid _tranID, XAResource _theXAResource, int xaState, int txInfoState) throws XAException {
+		List<XAResource> xars = new ArrayList<XAResource>();
```

Comment:
You could save declaring this memory (and the extra processing needed to add to and to iterate through the list) if you put the code from line 1048 through 1066 into a separate private method.